File: framework/languages/java/samples/java/Bingo/Server/Api/src/main/java/systems/zlink/samples/bingo/server/api/handlers/AuthenticatePlayerHandler.java
package systems.zlink.samples.bingo.server.api.handlers;

import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;
import systems.zlink.samples.bingo.shared.contracts.Messages;

@ZLinkHandlerGroup("api")
public final class AuthenticatePlayerHandler
    implements ZLinkRequestHandler<
        Messages.AuthenticatePlayerReq,
        Messages.AuthenticatePlayerRes> {
    @Override
    public Messages.AuthenticatePlayerRes handle(
        Messages.AuthenticatePlayerReq request,
        ZLinkRequestContext context) {
        String actorId = request.accessToken();
        return new Messages.AuthenticatePlayerRes(
            true,
            actorId,
            actorId,
            null);
    }
}

File: framework/languages/java/samples/kotlin/Bingo/Server/Api/src/main/kotlin/systems/zlink/samples/kotlin/bingo/server/api/handlers/AuthenticatePlayerHandler.kt
package systems.zlink.samples.kotlin.bingo.server.api.handlers

import systems.zlink.framework.channels.ZLinkRequestContext
import systems.zlink.framework.channels.ZLinkRequestHandler
import systems.zlink.framework.handlers.ZLinkHandlerGroup
import systems.zlink.framework.kotlin.ZLinkCoroutineRuntime
import systems.zlink.samples.kotlin.bingo.shared.contracts.AuthenticatePlayerReq
import systems.zlink.samples.kotlin.bingo.shared.contracts.AuthenticatePlayerRes

@ZLinkHandlerGroup("api")
class AuthenticatePlayerHandler(
    private val coroutines: ZLinkCoroutineRuntime,
) : ZLinkRequestHandler<AuthenticatePlayerReq, AuthenticatePlayerRes> {
    override fun handle(
        request: AuthenticatePlayerReq,
        context: ZLinkRequestContext,
    ) = coroutines.blocking {
        AuthenticatePlayerRes(
            accepted = true,
            actorId = request.accessToken,
            displayName = request.accessToken,
            reason = null,
        )
    }
}
